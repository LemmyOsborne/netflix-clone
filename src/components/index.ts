export { Jumbotron } from "./Jumbotron/Jumbotron";
export { Footer } from "./Footer/Footer";