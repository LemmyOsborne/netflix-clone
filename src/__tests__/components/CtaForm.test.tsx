import React from "react";
import { CtaForm } from "../../components";
import { render } from "@testing-library/react";

jest.mock("react-router-dom")

describe("<CtaFotm/>", () => {
    it("render <CtaForm/>", () => {
        const { container, getByText, getByPlaceholderText, getByTestId } = render(
            <CtaForm direction="row">
                <CtaForm.InputGroup data-testid="input-group">
                    <CtaForm.Text>Ready to watch? Enter your email to create or restart your membership.</CtaForm.Text>
                    <CtaForm.Input placeholder="Email address" />
                    <CtaForm.Button>Try it now</CtaForm.Button>
                </CtaForm.InputGroup>
            </CtaForm>
        )

        expect(getByText("Ready to watch? Enter your email to create or restart your membership.")).toBeTruthy()
        expect(getByText("Try it now")).toBeTruthy()
        expect(getByPlaceholderText("Email address")).toBeTruthy()
        expect(getByTestId("input-group")).toBeTruthy()
        expect(container.firstChild).toMatchSnapshot()
    })
})