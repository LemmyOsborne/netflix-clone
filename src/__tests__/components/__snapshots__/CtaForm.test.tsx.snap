// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CtaFotm/> render <CtaForm/> 1`] = `
<div
  class="sc-hGPBjI iTCSKB"
  direction="row"
>
  <div
    class="sc-dlVxhl lhvevl"
    data-testid="input-group"
  >
    <p
      class="sc-bBHxTw dCICuo"
    >
      Ready to watch? Enter your email to create or restart your membership.
    </p>
    <input
      class="sc-kfPuZi cbTujw"
      placeholder="Email address"
    />
    <button
      class="sc-fKVqWL bdthkQ"
    >
      Try it now
       
      <img
        alt="Try now"
        src="/images/icons/chevron-right.png"
      />
    </button>
  </div>
</div>
`;
